azurerm/internal/services/media/media_streaming_endpoint_resource.go

```diff
@@ -0,0 +1,591 @@
+		return fmt.Errorf("Error waiting for creation of Streaming Endpoint %q in Media Services Account %q (Resource Group %q): %+v", streamingEndpointName, accountName, resourceGroup, err)
+	}
+
+	endpoint, err := client.Get(ctx, resourceGroup, accountName, streamingEndpointName)
+	if err != nil {
+		return fmt.Errorf("Error retrieving Streaming Endpoint %q from Media Services Account %q (Resource Group %q): %+v", streamingEndpointName, accountName, resourceGroup, err)
+	}
+
+	d.SetId(*endpoint.ID)
+
+	return resourceMediaStreamingEndpointRead(d, meta)
+}
+
+func resourceMediaStreamingEndpointRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Media.StreamingEndpointsClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.StreamingEndpointID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resp, err := client.Get(ctx, id.ResourceGroup, id.MediaserviceName, id.Name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[INFO] Streaming Endpoint %q was not found in Media Services Account %q and Resource Group %q - removing from state", id.Name, id.MediaserviceName, id.ResourceGroup)
+			d.SetId("")
+			return nil
+		}
+
+		return fmt.Errorf("Error retrieving Streaming Endpoint %q in Media Services Account %q (Resource Group %q): %+v", id.Name, id.MediaserviceName, id.ResourceGroup, err)
+	}
+
+	d.Set("name", id.Name)
+	d.Set("resource_group_name", id.ResourceGroup)
+	d.Set("media_services_account_name", id.MediaserviceName)
+	if location := resp.Location; location != nil {
+		d.Set("location", azure.NormalizeLocation(*location))
+	}
+
+	if props := resp.StreamingEndpointProperties; props != nil {
+		if scaleUnits := props.ScaleUnits; scaleUnits != nil {
+			d.Set("scale_units", scaleUnits)
+		}
+
+		accessControl := flattenAccessControl(props.AccessControl)
+		if err := d.Set("access_control", accessControl); err != nil {
+			return fmt.Errorf("Error flattening `access_control`: %s", err)
+		}
+
+		if cdnEnabled := props.CdnEnabled; cdnEnabled != nil {
+			d.Set("cdn_enabled", cdnEnabled)
+		}
+
+		if cdnProfile := props.CdnProfile; cdnProfile != nil {
+			d.Set("cdn_profile", cdnProfile)
+		}
+
+		if cdnProvider := props.CdnProvider; cdnProvider != nil {
+			d.Set("cdn_provider", cdnProvider)
+		}
+
+		crossSiteAccessPolicies := flattenCrossSiteAccessPolicies(resp.CrossSiteAccessPolicies)
+		if err := d.Set("cross_site_access_policies", crossSiteAccessPolicies); err != nil {
+			return fmt.Errorf("Error flattening `cross_site_access_policies`: %s", err)
+		}
+
+		if customHostNames := props.CustomHostNames; customHostNames != nil {
+			d.Set("custom_host_names", customHostNames)
+		}
+
+		if description := props.Description; description != nil {
+			d.Set("description", description)
+		}
+
+		if maxCacheAge := props.MaxCacheAge; maxCacheAge != nil {
+			d.Set("max_cache_age_seconds", maxCacheAge)
+		}
```

Comment:
since this one will be `*int64` this needs to become:

```suggestion
		maxCacheAge := 0
		if props.MaxCacheAge != nil {
			maxCacheAge = int(*props.MaxCacheAge)
		}
		d.Set("max_cache_age_seconds", maxCacheAge)
```